webpack/assets/javascripts/react_app/components/charts/ChartModal.jsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import {Tooltip} from 'react-bootstrap';
+import {OverlayTrigger} from 'react-bootstrap';
+import {Modal} from 'react-bootstrap';
+import Component from '../common/BindingComponent';
+
+export default class extends Component {
+    constructor(props) {
+        super(props);
+        this.bind(
+            "init",
+            "onHide"
+        );
+    }
+    init() {
+        this.props.drawChart();
```

Comment:
props is a hash. it includes all the properties defined on the 'html' tag. In this case, the properties are defined in StatisticsChartBox.js. {...this.props} is syntactic sugar for all of the adding all of the parents' props.

                <ChartModal {...this.props}
                    show={this.state.showModal}
                    onHide={this.closeModal}
                    drawChart={this.drawModal}
                />